cfme/intelligence/reports/reports.py

```diff
@@ -78,8 +79,20 @@ class filter(WaitTab):  # noqa
         interval = BootstrapSelect("cb_interval")
         interval_size = BootstrapSelect("cb_interval_size")
         interval_end = BootstrapSelect("cb_end_interval_offset")
-        primary_filter = ExpressionEditor()
-        secondary_filter = ExpressionEditor()
+        primary_record_filter = Button("contains", "Record Filter")
+        secondary_display_filter = Button("contains", "Display Filter")
+
+        @View.nested
+        class primary_filter(ExpressionEditor):     # noqa
+            def child_widget_accessed(self, widget):
+                if self.parent.primary_record_filter.is_displayed:
+                    self.parent.primary_record_filter.click()
```

Comment:
Question: can this button be disabled? What is expected outcome of click() when its disabled?

Reply:
Nope, it will never be disabled. If we are not allowed to set a filter for the report, we wouldn't be allowed to navigate to that tab in the first place.